ProjFS.Mac/PrjFSKext/PrjFSKext/VirtualizationRoots.cpp

```diff
@@ -132,20 +132,25 @@ kern_return_t VirtualizationRoots_Cleanup()
     return KERN_FAILURE;
 }
 
-VirtualizationRootHandle VirtualizationRoot_FindForVnode(vnode_t _Nonnull vnode, const FsidInode& vnodeFsidInode)
+VirtualizationRootHandle VirtualizationRoot_FindForVnode(
+    PerfTracer* _Nonnull perfTracer,
+    PrjFSPerfCounter baseCounter,
+    vnode_t _Nonnull vnode,
+    const FsidInode& vnodeFsidInode)
 {
-//    ProfileSample functionSample(Probe_VirtualizationRoot_Find);
-
+    PerfSample findForVnodeSample(perfTracer, baseCounter);
+    
     VirtualizationRootHandle rootHandle = RootHandle_None;
     
     vnode_get(vnode);
     // Search up the tree until we hit a known virtualization root or THE root of the file system
     while (RootHandle_None == rootHandle && NULLVP != vnode && !vnode_isvroot(vnode))
     {
-//        ProfileSample iterationSample(Probe_VirtualizationRoot_FindIteration);
+        PerfSample iterationSample(perfTracer, (PrjFSPerfCounter)(baseCounter + 1));
```

Comment:
It took me some jumping around to grok that this is just picking the _Iteration variant of FileOp_FindRoot or VnodeOp_FindRoot. Please comment the intent.

Reply:
Yep I plan to clean this up. When I wrote this, I was uncertain that I would even keep this pattern. I do think it is useful, so I'll think about how to make this more maintainable.

Reply:
I got rid of the enum math here, and the caller now has to pass in two counters. Makes the pattern a lot cleaner, even if slightly more verbose.